plugins/auto-sizes/hooks.php

```diff
@@ -100,16 +119,64 @@ function auto_sizes_render_generator(): void {
  * @param array<string, mixed> $parsed_block The parsed block.
  * @return string The updated block content.
  */
-function auto_sizes_improve_image_sizes_attribute( string $content, array $parsed_block ): string {
-
+function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
 	$processor = new WP_HTML_Tag_Processor( $content );
 	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
 
 	// Only update the markup if an image is found.
 	if ( $has_image ) {
+		$align = $parsed_block['attrs']['align'] ?? 'default';
+
+		if ( $align ) {
+			$processor->set_attribute( 'data-align', $align );
+		}
```

Comment:
~Would `$align` ever be empty due to the `??`? If not, then this is simpler:~

```suggestion
		$align = parsed_block['attrs']['align'] ?? 'default';
		$processor->set_attribute( 'data-align', $align );
```

Reply:
Alternatively, I see that this `data-align` attribute is serving two purposes: (1) exposing the `align` block attribute to the HTML tag processor and (2) indicating whether the tag has been processed. What if instead the tag processor added a unique attribute like `data-needs-sizes-update` and that this attribute is then looked for when avoiding redundant optimization. So something like this:

```suggestion
		$processor->set_attribute( 'data-needs-sizes-update', true );

		$align = $parsed_block['attrs']['align'] ?? '';
		if ( $align ) {
			$processor->set_attribute( 'data-align', $align );
		}
```

And then see suggestions below.